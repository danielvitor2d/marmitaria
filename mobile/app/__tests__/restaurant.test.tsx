import { render } from "@testing-library/react-native";

import AuthContext, { UserType } from "../../src/contexts/auth";
import Restaurant from "../restaurant";

jest.mock("../../src/assets/profile.svg", () => "string qualquer");
jest.mock("../../src/assets/restaurant.png", () => "outra string qualquer");
jest.mock("../../src/assets/mini_profile.svg", () => "outra string qualquer");

describe("test login page", () => {
  let page: any;

  beforeEach(() => {
    page = render(
      <AuthContext.Provider
        value={{
          isAdmin: false,
          meal: null,
          rest: {
            address: '',
            id: '',
            isSuggestion: false,
            meals: [],
            name: '',
            paymentforms: '',
            value: '',
          },
          signed: true,
          user: {} as UserType,
          logout: async () => {},
          refetchUser: async () => {},
          setMeal: () => {},
          setRest: () => {},
          signIn: async () => ({ logged: true, isAdmin: false }),
          update: async () => true,
          setSuggestion: () => {},
          suggestion: {},
        }}
      >
        <Restaurant />
      </AuthContext.Provider>
    );
  });

  it("snapshot test", () => {
    expect(page).toMatchSnapshot();
  });

  it("Testando botão de voltar", () => {
    expect(page.getByTestId("goBack")).toBeDefined();
  });

  it("Testando botão de perfil", () => {
    expect(page.getByTestId("profile")).toBeDefined();
  });

  it("Testando botão de sugerir nova marmita", () => {
    expect(page.getByTestId("suggestNewMeal")).toBeDefined();
  });

  it("Testando lista de refeições", () => {
    expect(page.getByTestId("mealsList")).toBeDefined();
  });
});
