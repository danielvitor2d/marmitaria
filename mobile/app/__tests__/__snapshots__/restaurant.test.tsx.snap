// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test login page snapshot test 1`] = `
<View
  style={
    [
      {
        "flexBasis": "0%",
        "flexGrow": 1,
        "flexShrink": 1,
      },
      {
        "position": "relative",
      },
    ]
  }
>
  <View
    style={
      [
        {
          "width": "100%",
        },
        {
          "height": 56,
        },
        {
          "paddingLeft": 16,
          "paddingRight": 16,
        },
        {
          "backgroundColor": "#A60C0C",
        },
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "space-between",
        },
        {
          "alignItems": "center",
        },
        [
          {
            "position": "absolute",
          },
          {
            "top": 0,
          },
          {
            "left": 0,
          },
          {
            "right": 0,
          },
        ],
      ]
    }
    testID="headerTestId"
  >
    <View
      style={
        [
          {
            "color": "#fff",
          },
        ]
      }
      testID="goBack"
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="backButtonTestId"
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "white",
                "fontSize": 28,
              },
              undefined,
              {
                "fontFamily": "feather",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          ÔÑê
        </Text>
      </View>
    </View>
    <Text
      style={
        [
          {
            "color": "#fff",
          },
          {
            "fontSize": 18,
            "lineHeight": 28,
          },
        ]
      }
    >
      Marmitas
    </Text>
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "opacity": 1,
        }
      }
      testID="profile"
    >
      <outra string qualquer
        className="h-2 w-2 border bg-green-600"
      />
    </View>
  </View>
  <RCTScrollView
    style={
      [
        {
          "flexBasis": "0%",
          "flexGrow": 1,
          "flexShrink": 1,
        },
        {
          "marginTop": 64,
        },
      ]
    }
  >
    <View>
      <View
        style={
          [
            {
              "width": "100%",
            },
            {
              "height": 80,
            },
            {
              "backgroundColor": "#fff",
            },
            {
              "alignItems": "center",
            },
            {
              "justifyContent": "center",
            },
          ]
        }
      >
        <Text
          style={
            [
              {
                "fontSize": 20,
                "lineHeight": 28,
              },
              {
                "color": "#A60C0C",
              },
            ]
          }
        />
      </View>
      <View
        style={
          [
            {
              "paddingLeft": 4,
              "paddingRight": 4,
            },
            {
              "marginTop": 8,
            },
          ]
        }
      >
        <Image
          source="outra string qualquer"
          style={
            [
              {
                "width": "100%",
              },
            ]
          }
        />
        <View
          style={
            [
              {
                "marginTop": 8,
              },
            ]
          }
          testID="mealsList"
        />
        <View
          accessibilityState={
            {
              "busy": undefined,
              "checked": undefined,
              "disabled": undefined,
              "expanded": undefined,
              "selected": undefined,
            }
          }
          accessibilityValue={
            {
              "max": undefined,
              "min": undefined,
              "now": undefined,
              "text": undefined,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            {
              "alignItems": "center",
              "backgroundColor": "#A60C0C",
              "borderBottomLeftRadius": 6,
              "borderBottomRightRadius": 6,
              "borderTopLeftRadius": 6,
              "borderTopRightRadius": 6,
              "justifyContent": "center",
              "opacity": 1,
              "paddingBottom": 8,
              "paddingLeft": 16,
              "paddingRight": 16,
              "paddingTop": 8,
            }
          }
          testID="suggestNewMeal"
        >
          <Text
            style={
              [
                {
                  "fontSize": 20,
                  "lineHeight": 28,
                },
                {
                  "color": "#fff",
                },
              ]
            }
          >
            Sugerir nova marmita
          </Text>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
